app/modules/wifi.c

```diff
@@ -551,161 +555,215 @@ static int wifi_station_getbroadcast( lua_State* L ){
-  int auto_connect=0;
-  const char *ssid = luaL_checklstring( L, 1, &sl );
-  if (sl>32 || ssid == NULL)
-    return luaL_error( L, "ssid:<32" );
-  const char *password = luaL_checklstring( L, 2, &pl );
-  if ((pl!=0 && (pl<8 || pl>64)) || password == NULL)
-    return luaL_error( L, "pwd:0,8~64" );
-
-  if(lua_isnumber(L, 3))
+  char bssid[17];
+  if(get_flash_cfg) wifi_station_get_config_default(&sta_conf);
+    else wifi_station_get_config(&sta_conf);
+  if(sta_conf.ssid==0)
   {
-    auto_connect=luaL_checkinteger( L, 3 );;
-    if ( auto_connect != 0 && auto_connect != 1)
-    	return luaL_error( L, "wrong arg type" );
-  }
-  else if (lua_isstring(L, 3)&& !(lua_isnumber(L, 3)))
-  {
-	  lua_pushnil(L);
-	  lua_insert(L, 3);
-	  auto_connect=1;
-
+    lua_pushnil(L);
+      return 1;
   }
   else
   {
-    if(lua_isnil(L, 3))
-	   	return luaL_error( L, "wrong arg type" );
-	auto_connect=1;
+    if(lua_isboolean(L, 1) && lua_toboolean(L, 1)==true)
+    {
+      lua_newtable(L);
+      lua_pushstring(L, sta_conf.ssid);
+      lua_setfield(L, -2, "ssid");
+      lua_pushstring(L, sta_conf.password);
+      lua_setfield(L, -2, "pwd");
+      if(sta_conf.bssid_set==1)
+      {
+        c_sprintf(bssid, MACSTR, MAC2STR(sta_conf.bssid));
+        lua_pushstring( L, bssid);
+        lua_setfield(L, -2, "bssid");
+      }
+      return 1;
+    }
+    else
+    {
+      lua_pushstring( L, sta_conf.ssid );
+      lua_pushstring( L, sta_conf.password );
+      lua_pushinteger( L, sta_conf.bssid_set);
+      c_sprintf(bssid, MACSTR, MAC2STR(sta_conf.bssid));
+      lua_pushstring( L, bssid);
+      return 4;
+    }
   }
+}
 
-  if(lua_isnumber(L, 4))
-  {
-    sta_conf.bssid_set = 0;
-	c_memset(sta_conf.bssid, 0, 6);
-  }
-  else
-  {
-    if (lua_isstring(L, 4))
-	{
-	  const char *macaddr = luaL_checklstring( L, 4, &ml );
-	  luaL_argcheck(L, ml==17, 1, INVALID_MAC_STR);
-	  c_memset(sta_conf.bssid, 0, 6);
-	  ets_str2macaddr(sta_conf.bssid, macaddr);
-	  sta_conf.bssid_set = 1;
-	}
-	else
-	{
-	  sta_conf.bssid_set = 0;
-	  c_memset(sta_conf.bssid, 0, 6);
-	}
-  }
+// Lua: wifi.sta.getconfig()
+static int wifi_station_getconfig_current(lua_State *L)
+{
+  return wifi_station_getconfig(L, false);
+}
+
+// Lua: wifi.sta.getdefaultconfig()
+static int wifi_station_getconfig_default(lua_State *L)
+{
+  return wifi_station_getconfig(L, true);
+}
+
+// Lua: wifi.sta.config()
+static int wifi_station_config( lua_State* L )
+{
+  struct station_config sta_conf;
+  bool auto_connect=true;
+  bool save_to_flash=true;
+  size_t sl, pl, ml;
 
   c_memset(sta_conf.ssid, 0, 32);
   c_memset(sta_conf.password, 0, 64);
-  c_memcpy(sta_conf.ssid, ssid, sl);
-  c_memcpy(sta_conf.password, password, pl);
-
-  NODE_DBG(sta_conf.ssid);
-  NODE_DBG(" %d\n", sl);
-  NODE_DBG(sta_conf.password);
-  NODE_DBG(" %d\n", pl);
-  NODE_DBG(" %d\n", sta_conf.bssid_set);
-  NODE_DBG( MACSTR, MAC2STR(sta_conf.bssid));
-  NODE_DBG("\n");
+  c_memset(sta_conf.bssid, 0, 6);
```

Comment:
Please use `sizeof(sta_conf.bssid)` instead of hard coded array length (and on the 2 lines above) to keep the code generic.
There are further similar cases below....